### PR TITLE
Admin implementation,WIP

### PR DESCRIPTION
**Pull request by Denrig**

### CODE REVIEW

File: app/controllers/admin_base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class AdminBaseController < ApplicationController
+  layout 'admin'
+  before_action :admin?
+
+  def index; end
+
+  def admin?
+    unless current_user.admin
+      flash[:danger] = 'Only privilaged users can acces that page!'
```
Comment (VictorMotogna) [inline app/controllers/admin_base_controller.rb:9]: small typo here, should be _privileged_

File: app/controllers/products_carts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    action = params[:operation]
+    product_cart = ProductsCart.find_by(id: params[:id])
+    case action
+    when 'sub'
+      product_cart.quantity -= 1
+    when 'add'
+      product_cart.quantity += 1
+    end
+
+    product_cart.quantity.zero? ? product_cart.destroy : product_cart.save
```
Comment (VictorMotogna) [inline app/controllers/products_carts_controller.rb:40]: nice 👍

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
     if user&.authenticate(seession_params[:password])
-      reset_session
-      log_in user
-      flash[:success] = "Welcome #{user.name}!"
-      redirect_to user
+      if user.activated?
+        foraward_url = session[:forward_url]
+        reset_session
+        log_in user
+        params[:session][:remember_me] == '1' ? remember(user) : forget(user)
```
Comment (VictorMotogna) [inline app/controllers/sessions_controller.rb:13]: what does `'1'` mean and where does it come from? Maybe if you could try to use something else than `'1'` to note session status (like an enum, boolean, more suggestive string)

File: app/models/user.rb [hunk 4]
```diff
@@ -1,10 +1,83 @@
...
   has_one :cart
   has_secure_password
 
   # Validations
-  validates :password, presence: true, length: { minimum: 6 }
-  VALID_NAME_REGEX = /\A[a-zA-Z ]+\z/.freeze
   validates :name, presence: true, length: { minimum: 8, maximum: 50 }, format: { with: VALID_NAME_REGEX }
   validates :email, presence: true, length: { maximum: 255 }, format: { with: URI::MailTo::EMAIL_REGEXP  }, uniqueness: true
+  validates :password, presence: true, length: { minimum: 6 }, allow_nil: true
+  class << self
```
Comment (VictorMotogna) [inline app/models/user.rb:14]: a cool option for secure passwords is already implemented in the bcrypt lib: https://apidock.com/rails/v4.0.2/ActiveModel/SecurePassword/ClassMethods/has_secure_password

Your implementation is nice and it's very good that you implemented from scratch initially 👍

File: app/controllers/sessions_controller.rb (hunk 3 above)
Comment (Denrig) [reply app/controllers/sessions_controller.rb:13]: '1' is the value that input type=checkbox provide if it's selected

File: app/views/admin/products/index.html.haml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    %td.price Price
+    %td.action Action
+  - @products.each do |product|
+    %tr.product-item
+      = render 'shared/product_horizontal_item',product: product,product_price: product.price
+      %td.action
+        = link_to '',admin_product_url(product),method: 'delete',class:'glyphicon glyphicon-trash'
+        = link_to '',edit_admin_product_url(product),class:'glyphicon glyphicon-pencil'
+        = link_to '',new_admin_product_url,class:'glyphicon glyphicon-plus'
+
```
Comment (florinionce) [inline app/views/admin/products/index.html.haml:14]: extra lines here

File: app/views/layouts/_header.html.haml [hunk 6]
```diff
@@ -9,17 +9,29 @@
...
             %li.divider
-            %li= link_to "Log Out", logout_path , method: :delete
+            -if @current_user.admin?
+              %li
+                %span.glyphicon.glyphicon-wrench
+                = link_to "Admin Page", admin_root_url
+            %li
+              %span.glyphicon.glyphicon-log-out
+              = link_to "Log Out", logout_path , method: :delete
       -else
```
Comment (florinionce) [inline app/views/layouts/_header.html.haml:35]: missing space

File: app/views/layouts/_header.html.haml [hunk 7]
```diff
@@ -9,17 +9,29 @@
...
+              = link_to "My Cart", current_user.cart
+            %li
+              %span.glyphicon.glyphicon-user
+              = link_to "Profile", current_user
+            %li
+              %span.glyphicon.glyphicon-cog
+              = link_to "Settings", '#'
             %li.divider
-            %li= link_to "Log Out", logout_path , method: :delete
+            -if @current_user.admin?
```
Comment (florinionce) [inline app/views/layouts/_header.html.haml:28]: missing space

**Pull request merged**